EUSS Demand Flexibility measures

JieXiong9119 opened the pull request:
Pull request overview
---------------------

<!--- DESCRIBE PURPOSE OF THIS PULL REQUEST -->

 - Code review for demand flexibility measures and tests
 -- thermostat control load shed
 -- thermostat control load shift
 -- dispatch schedule generation (supplemental)

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes singularity image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [x] Upgrade Measures
 - [ ] Reporting Measures
 - [ ] Postprocessing

Author pull request checklist:
<!--- Add to this list or remove from it as applicable.  This is a simple templated set of guidelines. -->
 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Added tests for new measures
 - [x] Updated measure .xml(s)
 - [ ] Register values added to `comstock_column_definitions.csv`
 - [x] Both `options_lookup.tsv` files updated
 - [ ] 10k+ test run
 - [ ] Change documentation written
 - [x] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [x] Added 'See ComStock License' language to first two lines of each code file
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

JieXiong9119 committed:
dispatch schedule generation method implement

Commit:
create empty resource script

Commit:
add copyright

Commit:
misc

Commit:
add sample run timestep input arg

Commit:
add number of timesteps for sample sims

Commit:
moving individual methods out of run method

Commit:
new measure template

Commit:
Revert "moving individual methods out of run method"

This reverts commit 5f7ef242554e038521c3e9bcdb21310e5277d3f1.

Commit:
Revert "add number of timesteps for sample sims"

This reverts commit dc1abd2a60b7b52c4e3c07e3000a296e62bf761d.

Commit:
Revert "add sample run timestep input arg"

This reverts commit ea5c273e94f1cf3ccf0a5c1883633328d0312778.

Commit:
Revert "misc"

This reverts commit a8c944df9918bcc77a6ce44b0ca867bf6c3e2308.

Commit:
Revert "add copyright"

This reverts commit 13a48fa367ff66060e127bc91d7b15157cafa98d.

Commit:
calling ruby scripts under resources folder

Commit:
initial clean up

Commit:
indentation clean up

Commit:
misc

Commit:
cleaning + debugging outputs

JieXiong9119 committed:
fix peak generation function

Commit:
replacing weather file path to model

Commit:
replacing weather file path to model

Commit:
Update df_thermostat_control_load_shed_test.rb

Commit:
Update df_thermostat_control_load_shed_test.rb

Commit:
removing output_path input arg

Commit:
Update df_thermostat_control_load_shed_test.rb

Commit:
misc

Commit:
Update df_thermostat_control_load_shed_test.rb

Commit:
unit test template update

JieXiong9119 committed:
fix random seed

Commit:
Update df_thermostat_control_load_shed_test.rb

JieXiong9119 committed:
update run manager method

JieXiong9119 committed:
setpoint adjustment

Commit:
making direct changes to idf

JieXiong9119 committed:
Update measure and test

add part year simulation feature
add applicability check for building type

JieXiong9119 committed:
update measure

update applicability check on building type
leap year check

JieXiong9119 committed:
update measure

Commit:
Update options_lookup.tsv

JieXiong9119 committed:
update measure

add initial and final conditions

JieXiong9119 committed:
Add load shift measure

JieXiong9119 committed:
update peak generation function and test

JieXiong9119 committed:
clean up unnecessary files

JieXiong9119 committed:
Update measure.xml and options_lookup.tsv

JieXiong9119 committed:
Add 'See ComStock License` in test files

JanghyunJK commented on resources/measures/df_thermostat_control_load_shift/tests/df_thermostat_control_load_shift_test.rb:
```diff
@@ -0,0 +1,334 @@
+  #   assert(y_seed == y_seed_hardcoded)
+  # end
+
+  # def test_dispatch_gen_small_run
```
@JieXiong9119 I believe we want to include these tests to run?! or am i missing anything here

JieXiong9119 committed:
Merge branch 'main' into jx/demand-flexibility

JieXiong9119 committed:
update measure

reset model config after sampling or change of time step

JieXiong9119 committed:
update measure

JieXiong9119 committed:
clean test models

JieXiong9119 committed:
update sensitivity report measure

add option of ScheduleInterval thermostat schedule reading

JieXiong9119 committed:
Update dispatch_schedule_generation.rb

JieXiong9119 committed:
Update qoi_report measure

Add df outputs to report measure

JieXiong9119 committed:
update qoi_report measure and supporting measure

JieXiong9119 committed:
update qoi report var name

JieXiong9119 committed:
Update comstock_column_definitions.csv

JieXiong9119 committed:
Update comstock_column_definitions

JieXiong9119 committed:
update options_lookups.tsv

JieXiong9119 committed:
Merge branch 'main' into jx/demand-flexibility

JieXiong9119 committed:
update options_lookup.tsv

JieXiong9119 committed:
Merge branch 'main' into jx/demand-flexibility

JieXiong9119 committed:
Update applicability check

JieXiong9119 committed:
Merge branch 'main' into jx/demand-flexibility

JieXiong9119 committed:
update measure

fix applicability check

JieXiong9119 committed:
update measure

JieXiong9119 committed:
update dispatch_schedule_generation.rb

address dispatch window on last day exceed year end

JieXiong9119 committed:
update options for load shift

JieXiong9119 committed:
update options_lookup.tsv

JieXiong9119 committed:
update options_lookup

JieXiong9119 committed:
update loadshift measure

separate pre-cooling and pre-heating based on season

JieXiong9119 committed:
Update load shift measure for pre-cooling only

JieXiong9119 committed:
update load shift measure to address abnormal heating sp

JieXiong9119 committed:
update load shift measure

not change htg sp sch
take design day htg sp sch into reference to adjust clg sp sch

JieXiong9119 committed:
Merge branch 'main' into jx/demand-flexibility

JieXiong9119 committed:
Clean up df measures
